File: kotlin/day19_part1_2016.kt
import java.io.File

fun main(args: Array<String>) {
    val totalElves = readInput("input.txt")
    val winner = findWinningElf(totalElves)
    println(winner)
}

fun readInput(filename: String): Int {
    val totalElves = File(filename).readLines()[0].toInt()
    return totalElves
}

fun findWinningElf(totalElves: Int): Int {
    var highestPowerOfTwo = 1
    while (highestPowerOfTwo * 2 <= totalElves) {
        highestPowerOfTwo *= 2
    }
    return (totalElves - highestPowerOfTwo) * 2 + 1
}

File: r/day19_part1_2016.R

totalElves <- as.integer(readLines("input.txt"))
highestPowerOfTwo <- 1
while (highestPowerOfTwo * 2 <= totalElves) {
  highestPowerOfTwo <- highestPowerOfTwo * 2
}
winner <- (totalElves - highestPowerOfTwo) * 2 + 1
cat(winner)
